### PR TITLE
Uses java try with resources to let java close resource automatically.

### PR DESCRIPTION
**Pull request by rajadilipkolli**
This commit use java try with resources which is a feature of Java 7 and above which closes resource automatically.

### CODE REVIEW

File: spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/HeapdumpMvcEndpoint.java [hunk 1]
```diff
@@ -144,22 +144,14 @@ protected void handle(File heapDumpFile, HttpServletRequest request,
 		response.setContentType("application/octet-stream");
 		response.setHeader("Content-Disposition",
 				"attachment; filename=\"" + (heapDumpFile.getName() + ".gz") + "\"");
-		try {
-			InputStream in = new FileInputStream(heapDumpFile);
+		try (InputStream in = new FileInputStream(heapDumpFile)) {
 			try {
 				GZIPOutputStream out = new GZIPOutputStream(response.getOutputStream());
```
Comment (rajadilipkolli) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/HeapdumpMvcEndpoint.java]: Agrees, I will make necessary changes and publish again

Comment (rajadilipkolli) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/HeapdumpMvcEndpoint.java]: @vborcea can you please review the changes?

**Comment by vborcea** [comment]
Check the conflicts

**Comment by wilkinsona** [comment]
Thanks for the pull request. It's now in `master`.

**Comment by rajadilipkolli** [comment]
@wilkinsona thanks got accepting , also this taught me how to look more into code base by looking at your polish code commit. thanks for expanding my mindset

**Pull request closed without merging**